.github/workflows/integration_test_run.yaml

```diff
@@ -196,37 +188,19 @@ jobs:
           microk8s-addons: ${{ inputs.microk8s-addons }}
           channel: ${{ inputs.channel }}
           juju-channel: ${{ inputs.juju-channel }}
-      - name: Enable microk8s registry
-        if: ${{ inputs.provider == 'microk8s' && github.event.pull_request.head.repo.fork }}
-        run: |
-          sudo microk8s enable registry
-          sudo microk8s kubectl -n container-registry rollout status -w deployment/registry
       - name: Download all artifacts
         uses: actions/download-artifact@v3
-        if: ${{ github.event.pull_request.head.repo.fork }}
-      - name: Push images to microk8s registry
-        if: ${{ inputs.provider == 'microk8s' && github.event.pull_request.head.repo.fork }}
+        with:
+          path: ${{ inputs.working-directory }}
+      - name: Enable ingress and import images
+        if: ${{ inputs.provider == 'microk8s' }}
+        working-directory: ${{ inputs.working-directory }}
         run: |
           for image_name in $(echo '${{ inputs.images }}' | jq -cr '.[]'); do
-            docker load < ${image_name}/${image_name}.tar
-            docker push localhost:32000/${image_name}:latest
+            sudo microk8s ctr images import - < ${image_name}/${image_name}.tar
           done
-      - name: Configure GHCR in microk8s
-        if: ${{ inputs.provider == 'microk8s' && !github.event.pull_request.head.repo.fork }}
-        run: |
-          # Adding authentication for ghcr.io for containerd as per https://microk8s.io/docs/registry-private
-          # Note: containerd has to be restarted for the changes to take effect
-          # (https://github.com/containerd/cri/blob/master/docs/registry.md)
-          sudo su -c 'echo "
-          [plugins.\"io.containerd.grpc.v1.cri\".registry.configs.\"ghcr.io\".auth]
-          username = \"${{ github.actor }}\"
-          password = \"${{ secrets.GITHUB_TOKEN }}\"
-          " >> /var/snap/microk8s/current/args/containerd-template.toml'
-          sudo su -c 'systemctl restart snap.microk8s.daemon-containerd.service && microk8s status --wait-ready'
-      - name: Enable microk8s ingress
-        if: ${{ inputs.provider == 'microk8s' }}
-        run: |
           sudo microk8s enable ingress
+          sudo microk8s ctr images ls
```

Comment:
Minor nitpick, I'm wondering if the output of this might be huge, perhaps a grep for the images that we're looking for might help reduce the output size?

Reply:
We can in fact remove that. Thanks